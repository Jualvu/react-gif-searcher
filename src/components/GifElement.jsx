import PropTypes from 'prop-types'

export const GifElement = ({ title, url }) => {
    return (

        <div className='card'>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>

    );

}
GifElement.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
};