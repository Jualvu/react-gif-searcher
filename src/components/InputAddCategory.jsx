import { useState } from 'react';
import PropTypes from 'prop-types';


export const InputAddCategory = ({ onNewCategory }) => {

    const [inputText, setInputText] = useState('');

    //metodo para manejar nuevos caracteres en el input para mostrarlos
    const handleChangeNewCategory = (event) => {
        //event es un evento de html, que retorna datos tras una interaccion. dentro de todos los datos
        // esta el target.value que es el texto que se esta introduciendo en el input
        setInputText(event.target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        //si el input tiene menos de 1 caracter no se sube el formulario
        if (inputText.trim().length <= 1) return;

        //PASAR DE VUELTA AL COMPONENTE PADRE EL INPUTTEXT
        onNewCategory(inputText);
        setInputText('');

    }

    return (
        <form onSubmit={(event) => onSubmit(event)} aria-label='form'>

            <input
                type='text'
                value={inputText}
                placeholder='Buscar gifs'
                onChange={handleChangeNewCategory}
            />

        </form>

    );

}
InputAddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}