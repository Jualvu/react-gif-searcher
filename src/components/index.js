
export * from './InputAddCategory'
export * from './GifGrid'
export * from './GifElement'