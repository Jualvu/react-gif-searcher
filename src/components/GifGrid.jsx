import PropTypes from 'prop-types'

import { GifElement } from './GifElement';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ categoryName }) => {

    const { categoryGifs, isLoading } = useFetchGifs(categoryName);

    return (
        <>
            <h3> {categoryName} </h3>
            {
                isLoading ?
                    (<h2>Cargando...</h2>)
                    : null
            }
            <div className='card-grid'>
                {
                    categoryGifs.map((gif) => {
                        return (
                            <GifElement
                                key={gif.id}
                                title={gif.title}
                                url={gif.url} />
                        );

                    })
                }

            </div>

        </>
    );

}
GifGrid.propTypes = {
    categoryName: PropTypes.string.isRequired,
}
