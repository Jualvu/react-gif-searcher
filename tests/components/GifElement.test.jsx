import { render, screen } from '@testing-library/react';
import { GifElement } from '../../src/components/GifElement'

/* eslint-disable no-undef */
describe('Pruebas en componente GifElement', () => {

    const title = 'Satiama';
    const url = 'https://one-punch.com/saitama.jpg';

    test('Prueba match con Snapshot ', () => {
        const { container } = render(<GifElement title={title} url={url} />);
        expect(container).toMatchSnapshot();

    });

    test('Debe mostrar la imagen con el URL y el ALT indicado', () => {

        render(<GifElement title={title} url={url} />);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);

    });

});