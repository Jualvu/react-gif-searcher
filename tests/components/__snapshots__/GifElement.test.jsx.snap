// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente GifElement Prueba match con Snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Satiama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Satiama
    </p>
  </div>
</div>
`;
